pandas-stubs/core/indexes/base.pyi

```diff
@@ -52,37 +60,50 @@ class InvalidIndexError(Exception): ...
 
 _str = str
 
-class _IndexGetitemMixin(Generic[S1]):
-    @overload
-    def __getitem__(
-        self,
-        idx: slice
-        | np_ndarray_anyint
-        | Sequence[int]
-        | Index
-        | Series[bool]
-        | Sequence[bool]
-        | np_ndarray_bool,
-    ) -> Self: ...
-    @overload
-    def __getitem__(self, idx: int) -> S1: ...
-
-class Index(IndexOpsMixin, PandasObject):
+class Index(IndexOpsMixin[S1]):
     __hash__: ClassVar[None]  # type: ignore[assignment]
+    # overloads with additional dtypes
+    @overload
+    def __new__(  # type: ignore[misc]
+        cls,
+        data: Iterable[int | np.integer],
```

Comment:
One might assume this would work with np.ndarray - it doesn't: https://github.com/numpy/numpy/issues/24335

Reply:
Added overloads for ndarray